chcemvediet/apps/anonymization/anonymization.py

```diff
@@ -53,47 +53,57 @@
     u'ž': u'(?:z|ž)',
 }
 
-def generate_word_pattern(words):
+def generate_word_pattern(words, match_subwords):
     u"""
     Generates list of patterns, that matches slovak accent insensitive lowercase word. Each word is
-    captured in group.
+    captured in group. If ``match_subwords`` is True, pattern will match too subword.
     """
     patterns = []
+    template = u'({})' if match_subwords else u'(\\b{}\\b)'
     for word in words:
         if len(word) < WORD_SIZE_MIN:
             continue
-        p = u''.join([TRANSLATE_TABLE[c] if c in TRANSLATE_TABLE else re.escape(c) for c in word.lower()])
-        patterns.append(u'(\\b{}\\b)'.format(p))
+        p = u''
+        for c in word.lower():
+            if c in TRANSLATE_TABLE:
+                p += TRANSLATE_TABLE[c]
+            else:
+                p += re.escape(c)
+                if not c.isalnum() and match_subwords:
```

Comment:
@martinmacko47 Ak by sme chceli matchnut slovo "hornadolna" aj v texte, tak by som pridal este jeden flag ako parameter do funckie.

Reply:
@viliambalaz Flag `match_subwords` zatial staci. Ak sa tie patterny budu v buducnosti moc komplikovat, bude sa asi treba zamysliet a spravit to nejaky univerzalnejsie.